core/src/data/literal.rs

```diff
@@ -190,6 +197,28 @@ impl<'a> Literal<'a> {
         }
     }
 
+    pub fn bitwise_shift_left(&self, other: &Literal<'a>) -> Result<Literal<'static>> {
+        match (self, other) {
+            (Number(l), Number(r)) => match (l.to_i64(), r.to_i64()) {
+                (Some(l), Some(r)) => match (l << r).to_bigint() {
+                    Some(res) => Ok(Number(Cow::Owned(BigDecimal::new(res, 0)))),
+                    None => Err(LiteralError::IncompatibleBitOperation(
+                        l.to_string(),
+                        r.to_string(),
+                    )
+                    .into()),
+                },
+                _ => Err(LiteralError::UnreachableBinaryArithmetic.into()),
+            },
+            (Null, Number(_)) | (Number(_), Null) | (Null, Null) => Ok(Literal::Null),
+            _ => Err(LiteralError::IncompatibleBitOperation(
+                format!("{:?}", self),
+                format!("{:?}", other),
+            )
+            .into()),
+        }
+    }
```

Comment:
here we can remove the use of unreachable error.

still we need to think about the proper error names, but briefly we can re-arrange codes like this below.

```rust
match (self, other) {
    (Number(l), Number(r)) => {
        let l = l.to_i64().unwrap(); // non-integer -> error
        let r = r.to_u32().unwrap();
        let res = l.checked_shl(r).unwrap(); // overflow -> error
        Ok(Number(Cow::Owned(BigDecimal::from(res)))
    }
    (Null, Number(_)) | (Number(_), Null) | (Null, Null) => Ok(Literal::Null),
    // it would be better to have more explicit error name because it is caused by non-integer literals
    _ => Err(LiteralError::IncompatibleBitOperation(
        format!("{:?}", self),
        format!("{:?}", other),
    )
    .into()),
}
```

Reply:
<img width="255" alt="image" src="https://github.com/gluesql/gluesql/assets/2025065/e8c66d26-e610-449b-bd34-0901b2b51240">